modtools/intro-loader.lua

```diff
@@ -0,0 +1,160 @@
+        to the prefix, as in INTRO_MOUNTAIN for dwarves
+      - If the -rand argument is supplied, the filenames must each have a unique
+        number appended, as in INTRO_MOUNTAIN_2
+      - Further, for the -rand argument, usable numbers start at 1 and all
+	    numbers must be consecutive.
+      - The files must not have file extensions
+]====]
+
+
+local utils = require 'utils'
+
+
+validArgs = validArgs or utils.invert({
+    'help',
+    'v'
+    'now',
+    'folder',
+    'file',
+    'rand',
+    'entity',
+})
+
+local args = utils.processArgs({...}, validArgs)
+
+if args.help then
+    print(help)
+    return
+end
+
+local inputfolder = dfhack.getDFPath() .. "\\data\\announcement\\"
+if args.folder then
+    inputfolder = dfhack.getDFPath() ..  args.folder .. "\\"
+end
+
+local prefix = "INTRO"
+if args.file then
+    prefix = args.file
+end
+
+if args.rand then
+    math.randomseed(os.time())
+end
+
+
+--Build current index of files in directory
+do
+    local OS = dfhack.getOSType()
+    if OS == 'windows' then
+        os.execute("dir \"" .. inputfolder .. "\" /L /B >index.txt")
+    elseif OS == 'linux' or OS == 'darwin' then --apple's real name is darwin, apparently
+        os.execute("ls \"" .. inputfolder .. "\" -1 -A >index.txt")
+    end
+    
+    local list = io.open(dfhack.getDFPath() .. "\\data\\announcement\\index.txt","r")
+    indextext = list:read("*all")
+    io.close(list)
+end
+
+
+--Code main body, this is where the magic happens
+local function introloader(folder,dorand,doent,verbose)
+    local entity = "_" .. df.historical_entity.find(df.global.ui.civ_id).entity_raw.code
+    if ~doent then entity = "" end
+    local inputfile = folder .. "\\" .. prefix .. entity
+    
+    if dorand then
+        --Determine number of usable files
+        local filecount = 0
+        while string.gmatch(indextext,inputfile .. "_%d+") do
+            filecount++
+        end
+        
+        --Random decision
+        if filecount > 0 then
+            local rand = math.random(1,filecount)
+            inputfile = inputfile .. "_" .. rand
+        end
+    end
+    
+    if verbose then print("Loading intro text") end
+    local input = assert(io.open(inputfile, "rb"),"No matching intro file, leaving as-is")
+    if input == false then return end
+    local output = assert(io.open(dfhack.getDFPath() .. "\\data\\announcement\\fortressintro", "wb"),"Could not create or open data/announcement/fortressintro, somehow.")
+    if output == false then input:close() return end
+    local t = input:read("*all")
+    t = string.gsub(t, "\r\n", "\n")
+    output:write(t)
+    if verbose then print("Civilization intro text written") end
+    output:close()
+    input:close()
+end
+
+
+
+--Flow control for the code; runs process code after embark selection has been made, to ensure proper operation.
+dfhack.onStateChange.loadNewFortressIntro = function(code)
+if(dorand|doent)
```

Comment:
I was having some issues really reading through processArgs(), honestly. Can you possibly explain what its returns really are?
The arguments being passed to the function in those variables should be getting derived from the work processArgs() is doing; the only reason they're named that there is because I'm not quite used to lua and its very global variables and their scope yet; I'm _pretty_ sure I don't even need to pass them into the function like I did.
Also, I clearly need to stop using bitwise operators.